fabric-entity-events-v1/src/main/java/net/fabricmc/fabric/api/entity/event/v1/ServerPlayerEvents.java

```diff
@@ -69,6 +85,19 @@
 		void afterRespawn(ServerPlayerEntity oldPlayer, ServerPlayerEntity newPlayer, boolean alive);
 	}
 
+	@FunctionalInterface
+	public interface BeforeDeath {
+		/**
+		 * Called when a player takes fatal damage.
+		 *
+		 * @param player the player
+		 * @param source the fatal damage source
+		 * @param amount the amount of damage that has killed the player
+		 * @return whether or not the player should die
+		 */
+		boolean beforeDeath(ServerPlayerEntity player, DamageSource source, float amount);
```

Comment:
`damage` would be a nicer and more descriptive name for `amount`

Reply:
It seems a bit weird to have `source` but also `damage` - maybe `damageSource` and `damageAmount`?

Reply:
Yeah why not have `damageSource` and `damageAmount`.

Reply:
Done